server/src/main/java/org/elasticsearch/action/admin/cluster/node/reload/NodesReloadSecureSettingsRequest.java

```diff
@@ -19,22 +19,96 @@
 
 package org.elasticsearch.action.admin.cluster.node.reload;
 
+import org.elasticsearch.Version;
 import org.elasticsearch.action.support.nodes.BaseNodesRequest;
+import org.elasticsearch.common.CharArrays;
+import org.elasticsearch.common.Nullable;
+import org.elasticsearch.common.bytes.BytesArray;
+import org.elasticsearch.common.bytes.BytesReference;
+import org.elasticsearch.common.io.stream.StreamInput;
+import org.elasticsearch.common.io.stream.StreamOutput;
+import org.elasticsearch.common.settings.SecureString;
+
+import java.io.IOException;
+import java.util.Arrays;
 
 /**
- * Request for a reload secure settings action.
+ * Request for a reload secure settings action
  */
 public class NodesReloadSecureSettingsRequest extends BaseNodesRequest<NodesReloadSecureSettingsRequest> {
 
+    /**
+     * The password is used to re-read and decrypt the contents
+     * of the node's keystore (backing the implementation of
+     * {@code SecureSettings}).
+     */
+    @Nullable
+    private SecureString secureSettingsPassword;
+
     public NodesReloadSecureSettingsRequest() {
     }
 
     /**
-     * Reload secure settings only on certain nodes, based on the nodes IDs specified. If none are passed, secure settings will be reloaded
-     * on all the nodes.
+     * Reload secure settings only on certain nodes, based on the nodes ids
+     * specified. If none are passed, secure settings will be reloaded on all the
+     * nodes.
      */
-    public NodesReloadSecureSettingsRequest(final String... nodesIds) {
+    public NodesReloadSecureSettingsRequest(String... nodesIds) {
         super(nodesIds);
     }
 
+    @Nullable
+    public SecureString getSecureSettingsPassword() {
+        return secureSettingsPassword;
+    }
+
+    public NodesReloadSecureSettingsRequest setSecureStorePassword(SecureString secureStorePassword) {
+        this.secureSettingsPassword = secureStorePassword;
+        return this;
+    }
+
+    public void closePassword() {
+        if (this.secureSettingsPassword != null) {
+            this.secureSettingsPassword.close();
+        }
+    }
+
+    boolean hasPassword() {
+        return this.secureSettingsPassword != null && this.secureSettingsPassword.length() > 0;
+    }
+
+    @Override
+    public void readFrom(StreamInput in) throws IOException {
+        super.readFrom(in);
+        if (in.getVersion().onOrAfter(Version.V_8_0_0)) {
```

Comment:
Isn't this intended to go back to 7.x? This should be the version we are backporting to, and then the bwc tests enabled in the root build.gradle while the backport is being done

Reply:
My plan of action was to merge this with `V_8_0_0` in master and change this to `Version.V_7_4_0` on backporting to `7.x`/`7.4`, and then change it to `Version.V_7_4_0` in master too

Reply:
That is generally a more error prone approach. It presents the possibility of forgetting a place that needs to change in the followup master commit. By putting the intended version into the original commit, we dont' run this risk. The procedure I suggested is the entire reason we have the bwc_tests_enabled flag in gradle.